src/types/excelsheet.rs

```diff
@@ -141,93 +142,21 @@ impl ExcelSheet {
     }
 }
 
-fn create_boolean_array(
-    data: &Range<CalDataType>,
-    col: usize,
-    offset: usize,
-    limit: usize,
-) -> Arc<dyn Array> {
-    Arc::new(BooleanArray::from_iter((offset..limit).map(|row| {
-        data.get((row, col)).and_then(|cell| cell.get_bool())
-    })))
-}
-
-fn create_int_array(
-    data: &Range<CalDataType>,
-    col: usize,
-    offset: usize,
-    limit: usize,
-) -> Arc<dyn Array> {
-    Arc::new(Int64Array::from_iter(
-        (offset..limit).map(|row| data.get((row, col)).and_then(|cell| cell.get_int())),
-    ))
-}
-
-fn create_float_array(
-    data: &Range<CalDataType>,
-    col: usize,
-    offset: usize,
-    limit: usize,
-) -> Arc<dyn Array> {
-    Arc::new(Float64Array::from_iter((offset..limit).map(|row| {
-        data.get((row, col)).and_then(|cell| cell.get_float())
-    })))
-}
-
-fn create_string_array(
-    data: &Range<CalDataType>,
-    col: usize,
-    offset: usize,
-    limit: usize,
-) -> Arc<dyn Array> {
-    Arc::new(StringArray::from_iter((offset..limit).map(|row| {
-        data.get((row, col)).and_then(|cell| cell.get_string())
-    })))
-}
-
 fn duration_type_to_i64(caldt: &CalDataType) -> Option<i64> {
     caldt
         .as_time()
         .map(|t| 1000 * i64::from(t.num_seconds_from_midnight()))
 }
 
-fn create_date_array(
-    data: &Range<CalDataType>,
-    col: usize,
-    offset: usize,
-    limit: usize,
-) -> Arc<dyn Array> {
-    let epoch = NaiveDate::from_ymd_opt(1970, 1, 1).unwrap();
-    Arc::new(Date32Array::from_iter((offset..limit).map(|row| {
-        data.get((row, col))
-            .and_then(|caldate| caldate.as_date())
-            .and_then(|date| i32::try_from(date.signed_duration_since(epoch).num_days()).ok())
-    })))
-}
-
-fn create_datetime_array(
+fn create_arrow_array<NT, T: ArrowPrimitiveType<Native = NT>, F: Fn(&CalDataType) -> Option<NT>>(
     data: &Range<CalDataType>,
     col: usize,
     offset: usize,
     limit: usize,
+    transformer: &F,
```

Comment:
Is borrowing the function really mandatory ?

Reply:
Yup. Since it get captured by a closure, it'd be moved if we received it by value